Add secure infra docs (#1184)
hashicorp/web-unified-docs: content/well-architected-framework/docs/docs/secure-systems/infrastructure/secure-access.mdx

---
page_title: Secure human access to infrastructure
description: Learn how to secure human access to infrastructure with modern identity management, ephemeral credentials, and zero-trust security.
---

# Secure human access to infrastructure

The move to cloud-native and ephemeral resources has changed how
people think about securing access to infrastructure. Regardless of your
deployment methodology, your teams still need to access infrastructure for
troubleshooting and incident response.

In traditional infrastructure deployments, you might have used bastion hosts or VPNs
to provide secure access to infrastructure. However, these methods can introduce
additional attack surfaces and may not align with modern security best
practices. Managing access to infrastructure typically relied on long-lived,
static credentials. Long-lived credentials increase the risk of compromise,
leading to unauthorized access.

## Why secure human access to infrastructure?

Securing human access to infrastructure is critical to protect against
unauthorized access and potential threats. Organizations must implement a strong

> identity and access management program, including:

Reviewer: ```suggestion
Securing human access to infrastructure is critical to protect against
unauthorized access and potential threats. Organizations must implement a strong
identity and access management program, including the following concepts:
```

  authorization:**](/well-architected-framework/secure-systems/identity-access-management/centralize-identity-management)
  to protect against unauthorized access and simplify identity management.
- [**Use strong authentication
  methods:**](/well-architected-framework/secure-systems/identity-access-management/implement-strong-sign-in-workflows)
  to protect against credential theft and unauthorized access.
- [**Switch to ephemeral, dynamic
  credentials:**](/well-architected-framework/secure-systems/identity-access-management/use-dynamic-credentials)
  to reduce the risk of credential compromise. 
- [**Audit and manage account
  access:**](/well-architected-framework/secure-systems/identity-access-management/manage-access-lifecycle)
  to ensure access is reviewed and revoked when no longer needed.

Implementing an effective identity and access management program sets the
foundation to secure human access to infrastructure.

## Deploy access controls with Terraform

HashiCorp Terraform helps you adapt to how you manage and audit infrastructure
access. Instead of platform engineers accessing infrastructure directly, they
use Terraform to define and manage infrastructure as code. 

> - **Eliminates direct console access** - Teams interact with infrastructure through code review and approval processes rather than direct cloud console access

Reviewer: ```suggestion
- **Eliminates direct console access:** Teams interact with infrastructure through code review and approval processes rather than direct cloud console access
```

> - **Enforces change management** - All infrastructure modifications must go through version control, peer review, and approval workflows

Reviewer: ```suggestion
- **Enforces change management:** All infrastructure modifications must go through version control, peer review, and approval workflows
```

- **Provides complete audit trails** - Every infrastructure change is documented in Git history with author, timestamp, and approval records

> - **Enables policy enforcement** - Terraform Cloud/Enterprise can enforce organization-wide security policies that prevent non-compliant infrastructure deployment

Reviewer: ```suggestion
- **Enables policy enforcement:** HCP Terraform and Terraform Enterprise can enforce organization-wide security policies that prevent non-compliant infrastructure deployment
```

  ip_configuration {
    name                          = "testconfiguration1"
    subnet_id                     = "/subscriptions/YOUR-SUBSCRIPTION-ID/resourceGroups/existing-rg/providers/Microsoft.Network/virtualNetworks/existing-vnet/subnets/existing-subnet"
    private_ip_address_allocation = "Dynamic"
  }
}
```

Once the team approves the changes, and ensures the changes align with their
organizational security and compliance policies, the platform engineer runs Terraform
to apply the changes to the infrastructure, or uses a GitOps workflow to deploy
the configuration once you approve and merge the changes. Using Terraform provides a
clear audit trail of who made changes to the infrastructure and when, without
the need for direct access to the infrastructure or platform itself.

For larger platform teams, you can also use HCP Waypoint to enable developer
self-service for common provisioning tasks. Waypoint provides a self-service
workflow for application and infrastructure deployments. Waypont abstracts the
infrastructure complexity away from developers, allowing them to deploy
applications without needing direct access to the underlying infrastructure.

> 

Reviewer: Can we add an intro sentence here?

- **Integrates with existing CI/CD** - Works with Git workflows to enable deployment without manual infrastructure interaction
- **Provides deployment visibility** - Teams can track application deployments without needing infrastructure monitoring access

Using Waypoint further reduces the number of people who need direct
infrastructure access, supporting the principle of least privilege while also
improving developer productivity.

## Secure remote access with Boundary

HashiCorp Boundary is a modern access management solution that provides secure access to
infrastructure without the need for bastion hosts or VPNs.

- **Eliminates SSH key distribution** - No need to manage, rotate, or secure SSH private keys across teams
- **Session-based access** - Creates temporary, auditable connections that automatically expire
- **Multi-hop architecture** - Securely access private resources without exposing network topology or requiring VPN infrastructure
- **Integration with identity providers** - Leverages existing Active Directory, LDAP, or cloud identity services for authentication

> - **Dynamic target discovery** - Automatically discovers and provides access to infrastructure resources based on user permissions

Reviewer: Ditto on the hyphens here
